core/src/main/java/de/bwaldvogel/mongo/oplog/OplogDocument.java

```diff
@@ -0,0 +1,130 @@
+package de.bwaldvogel.mongo.oplog;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import de.bwaldvogel.mongo.bson.Bson;
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+import de.bwaldvogel.mongo.bson.Document;
+
+public class OplogDocument implements Bson {
+    private Document document;
+
+    public OplogDocument() {
+        document = new Document();
+        withProtocolVersion(2L);
+    }
+
+    public Document toDocument() {
```

Comment:
if you return a **modifiable** view of your object prefer to prefix it with `as` (`asDocument` in this case)

To me, `to` prefix means a potentially non-modifiable copy of current instance while `as` allows to change current object. 

```java
asDocument().clear(); // current object modified
// vs 
toDocument().clear(); // current object remains unchanged only copy modified
```